src/main/scala/org/apache/spark/deploy/armada/Config.scala

```diff
@@ -58,36 +59,35 @@ private[spark] object Config {
   // See https://kubernetes.io/docs/concepts/overview/working-with-objects/labels/#syntax-and-character-set
   // Clients do not use the prefix, therefore we just accept the name portion on label selector names.
   private val label = "([\\w&&[^-_.]]([\\w-_.]{0,61}[\\w&&[^-_.]])?)"
-  private val labelSelectors: Regex = (s"^($label=$label(,$label=$label)*)?$$").r
+  private val labelSelectors: Regex = s"^($label=$label(,$label=$label)*)?$$".r
 
   private[armada] def selectorsValidator(selectors: CharSequence): Boolean = {
-    val selectorsMaybe = labelSelectors.findPrefixMatchOf(selectors)
-    selectorsMaybe match {
-      case Some(_) => true
-      case None => false
-    }
+    labelSelectors.findPrefixMatchOf(selectors).isDefined
+  }
+
+  private def parseCommaSeparatedK8sLabels(str: String): Seq[Try[(String, String)]] = {
+    str.trim.split(",").map(_.trim)
+      .filter(_.nonEmpty)
+      .map { entry =>
+        entry.trim.split("=", 2).map(_.trim) match {
+          case Array(key, value) if K8sLabelValidator.isValidKey(key) && K8sLabelValidator.isValidValue(value) =>
+            Success(key -> value)
+          case _ =>
+            Failure(new IllegalArgumentException(
+              s"Invalid selector format: '$entry' (expected key=value)"
+            ))
+        }
+      }
```

Comment:
This allows to reuse code by  `commaSeparatedLabelsToMap`  and `k8sLabelListValidator`.

Reply:
Probably better to merge your PR first.

I have ideas for some improvements, but you already implemented some of them.